Content.Client/UserInterface/Screens/SeparatedChatGameScreen.xaml.cs

```diff
@@ -36,10 +60,53 @@ private void ResizeActionContainer()
         Actions.ActionsContainer.MaxGridWidth = ViewportContainer.Size.X - indent;
     }
 
+    private void RecalculateViewportDesiredSize()
+    {
+        // When the main window is reszied, calculate a new min/max for the
+        // ViewportContainer so that the chat will prefer to resize instead
+        // of letterboxing the viewport.
+        var uiScale = _cfg.GetCVar(CVars.DisplayUIScale);
+        if (uiScale == 0f)
+            uiScale = _uiManager.DefaultUIScale;
+
+        // CurrentRenderScale would be more correct, but it gets updated after us.
+        var gameScale = MainViewport.Viewport.FixedRenderScale;
+        var minFromVp = EyeManager.PixelsPerMeter * gameScale * _cfg.GetCVar(CCVars.ViewportMinimumWidth) / uiScale;
+        var maxFromVp = EyeManager.PixelsPerMeter * gameScale * _cfg.GetCVar(CCVars.ViewportMaximumWidth) / uiScale;
+        // Inventory bar also supplies a minWidth such that body+shoes don't overlap PDA.
+        var minFromUi = 1060;
+        // In case of overscaled viewport, chat hard-minWidth overpowers them both.
+        var minFromChat = Width - _chatMinWidth;
+
+        var min = Math.Min(Math.Max(minFromVp, minFromUi), minFromChat);
+        var max = Math.Max(maxFromVp, min);
+
+        // SplitContainer doesn't respect MaxSize, so set a MinSize on the chat instead.
+        ViewportContainer.MinWidth = min;
+        SeparatedChatPanel.MinWidth = Math.Max(_chatMinWidth, Width - max);
+
+        if (_deferredSplitFraction is float fraction)
```

Comment:
We generally avoid directly typing things if we don't have to, and this is functionally equivalent.

```suggestion
        if (_deferredSplitFraction is { } fraction)
```